File: src/RussianSitesStatus/Models/ViewModels/Statistic.cs
namespace RussianSitesStatus.Models;

public class Statistic
{
    public Navigation Navigation { get; set; }
    public List<Period> Periods { get; set; }
    public List<Data> Data { get; set; }
}

public class Data
{
    public int Up { get; set; }
    public int Down { get; set; }
    public int Unknown { get; set; }
    public string Label { get; set; }
}

public class Period
{
    public string Name { get; set; }
    public bool Current { get; set; }
    public string Url { get; set; }
}

public class Navigation
{
    public Link Current { get; set; }
    public Link Next { get; set; }
    public Link Prev { get; set; }
}

public class Link
{
    public string Name { get; set; }
    public string Url { get; set; }
}

public enum PeriodType
{
    Hour, 
    Day, 
    Week, 
    Month
}

File: UI/src/app/models/statistic.ts
export class StatisticVm {
    navigation: Navigation;
    periods: Period[];
    data: Data;
}

export class Data {
    uptime: number;
    history: Span[];
}

export class Span {
    up: number;
    down: number;
    unknown: number;
    label: string;
}

export class Period {
    name: string;
    current: boolean;
    url: string;
}

export class Navigation {
    current: Link;
    next: Link;
    prev: Link;
}

export class Link {
    name: string;
    url: string;
}

export enum PeriodType {
    Hour,
    Day,
    Week,
    Month
}